checker.go

```diff
@@ -109,41 +110,12 @@ func (c *cmpEqualsChecker) Check(got interface{}, args []interface{}, note func(
 //
 var DeepEquals = CmpEquals()
 
-// ContentEquals is a Checker checking that the two provided slices have the
-// same items, without respect to order.
-// For instance:
-//
-//     c.Assert([]int{1, 2, 3], qt.ContentEquals, []int{3, 2, 1})
-//
-var ContentEquals Checker = &contentEqualsChecker{
-	checker: CmpEquals(cmpopts.SortSlices(func(x, y interface{}) bool {
-		// TODO frankban: implement a proper sort function.
-		return pretty.Sprint(x) < pretty.Sprint(y)
-	})),
-	info: newInfo("content equals", "got", "want"),
-}
-
-type contentEqualsChecker struct {
-	checker Checker
-	info
-}
-
-// Check implements Checker.Check by checking that, got and args[0] are slices
-// of the same type, and they contain the same items, without respect to order.
-func (c *contentEqualsChecker) Check(got interface{}, args []interface{}, note func(key, value string)) error {
-	want := args[0]
-	gotType, wantType := reflect.TypeOf(got), reflect.TypeOf(want)
-	if gotType.Kind() != reflect.Slice {
-		return BadCheckf("got value should be a slice, but it is a %s instead", gotType.String())
-	}
-	if wantType.Kind() != reflect.Slice {
-		return BadCheckf("want value should be a slice, but it is a %s instead", wantType.String())
-	}
-	if gotElem, wantElem := gotType.Elem(), wantType.Elem(); gotElem != wantElem {
-		return BadCheckf("values are not slices of the same type: []%s != []%s", gotElem.String(), wantElem.String())
-	}
-	return c.checker.Check(got, args, note)
-}
+// ContentEquals is like DeepEquals but any slices in the compared values will
+// be sorted before being compared.
+var ContentEquals = CmpEquals(cmpopts.SortSlices(func(x, y interface{}) bool {
+	// TODO frankban: implement a proper sort function.
+	return pretty.Sprint(x) < pretty.Sprint(y)
```

Comment:
Does pretty print maps in a consistent order?

Reply:
(if not, then using ContentEquals on a slice of maps may panic, I think)

Reply:
This is a very good point, pretty on maps is not deterministic. We will handle that in a follow-up branch.